pilot/pkg/serviceregistry/kube/controller/discoverycontrollers.go

```diff
@@ -54,6 +55,12 @@ func (c *Controller) initDiscoveryNamespaceHandlers(
 			if discoveryNamespacesFilter.NamespaceCreated(ns.ObjectMeta) {
 				c.queue.Push(func() error {
 					c.handleSelectedNamespace(endpointMode, ns.Name)
+					if features.EnableEnhancedResourceScoping {
+						c.opts.XDSUpdater.ConfigUpdate(&model.PushRequest{
```

Comment:
why do we need to push for a new namespace? isn't there nothing in the namespace to begin with anyways?

Reply:
>why do we need to push for a new namespace? isn't there nothing in the namespace to begin with anyways?

Look into it, deeper I agree with you, I did this just to keep consistent with `handleSelectedNamespace`. Seems handleSelectedNamespace is also redundant

Reply:
Ah, the reason it was here is because namespace handler can take sometime.

If removed TestController_ServiceWithFixedDiscoveryNamespaces flakes

Reply:
I'm not sure if triggering a full push is enough
For example ServiceEntry has its own controller for managing state, I think we should trigger events for SE(like service/pod/endpoints) as well to make the controller reconcile

Reply:
I think so, need to do that in both handle selected and deselected function

Reply:
@dddddai Added